smithy-model/src/main/java/software/amazon/smithy/model/traits/TraitDefinition.java

```diff
@@ -158,12 +157,6 @@ public Builder selector(Selector selector) {
 
         public Builder addConflict(String trait) {
             Objects.requireNonNull(trait);
-
-            // Use absolute trait names.
-            if (!trait.contains("#")) {
```

Comment:
I consider this a bug since it's at odds with how literally every other value is resolved to a shape ID. There were two instances that used this in the prelude that were fixed. I did not see any other cases that would be affected by this in models that I sampled.